### PR TITLE
S6R1 Gungame stuff

### PR DESCRIPTION
**Pull request by SuicidalInsanity**

### CODE REVIEW

File: BDArmory/Competition/BDACompetitionMode.cs [hunk 1]
```diff
@@ -118,6 +131,16 @@ void Awake()
         void Start()
         {
             UpdateGUIElements();
+            headshotClip = SoundUtils.GetAudioClip("BDArmory/Sounds/Announcer/Headshot");
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Competition/BDACompetitionMode.cs:134]: Do we want to leave announcer functionality in BDA? I could just remove it entirely and save us the trouble.

[the remark this answers is not in the document]
Comment (BrettRyland) [reply BDArmory/Competition/BDACompetitionMode.cs:134]: That's definitely an option, yeah. Opinions in loggins seemed mixed about them.

File: BDArmory/Weapons/ModuleWeapon.cs [hunk 2]
```diff
@@ -1902,11 +1902,11 @@ void FixedUpdate()
                 }
                 else if (eWeaponType == WeaponTypes.Laser)
                 {
-                    for (int i = 0; i < laserRenderers.Length; i++)
-                    {
-                        laserRenderers[i].enabled = false;
-                    }
-                    //audioSource.Stop();
+                        for (int i = 0; i < laserRenderers.Length; i++)
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/Weapons/ModuleWeapon.cs:1905]: No. VS being VS, I guess.

File: BDArmory/VesselSpawning/SpawnUtils.cs [hunk 3]
```diff
@@ -16,6 +16,7 @@
 using BDArmory.Weapons;
 using BDArmory.Damage;
 using BDArmory.FX;
+using static UnityEngine.GUI;
```
[the remark this answers is not in the document]
Comment (SuicidalInsanity) [reply BDArmory/VesselSpawning/SpawnUtils.cs:19]: Visual Studio!!

**Review by BrettRyland — COMMENTED** [review]
Here's a diff with all the changes I mentioned and have made locally:
```diff
diff --git a/BDArmory/Competition/BDACompetitionMode.cs b/BDArmory/Competition/BDACompetitionMode.cs
index 8de0bfd65..0f93e9734 100644
--- a/BDArmory/Competition/BDACompetitionMode.cs
+++ b/BDArmory/Competition/BDACompetitionMode.cs
@@ -140,7 +140,7 @@ void Start()
             mmmonsterKillClip = SoundUtils.GetAudioClip("BDArmory/Sounds/Announcer/MonsterKill", true);
             ludicrousKillClip = SoundUtils.GetAudioClip("BDArmory/Sounds/Announcer/Godlike", true);
             audioSource = gameObject.AddComponent<AudioSource>();
-            announcerBarks = new List<AudioClip> { doubleKillClip, tripleKillClip, MultiKillClip, megaKillClip, ultraKillClip, mmmonsterKillClip, ludicrousKillClip };
+            announcerBarks = [doubleKillClip, tripleKillClip, MultiKillClip, megaKillClip, ultraKillClip, mmmonsterKillClip, ludicrousKillClip];
         }
 
         void OnGUI()
@@ -2886,7 +2886,6 @@ public void DoUpdate()
                                 {
                                     statusMessage += Scores.ScoreData[player].lastPersonWhoDamagedMe + " (NAILED 'EM! CLEAN KILL!)";
                                 }
-                                PlayAnnouncer(Scores.ScoreData[Scores.ScoreData[player].lastPersonWhoDamagedMe].killsThisLife, false, Scores.ScoreData[player].lastPersonWhoDamagedMe);
                                 //canAssignMutator = true;
                                 break;
                             case AliveState.HeadShot: // Damaged recently, but took damage a while ago from someone else.
@@ -2898,7 +2897,6 @@ public void DoUpdate()
                                 {
                                     statusMessage += Scores.ScoreData[player].lastPersonWhoDamagedMe + " (BOOM! HEAD SHOT!)";
                                 }
-                                PlayAnnouncer(Scores.ScoreData[Scores.ScoreData[player].lastPersonWhoDamagedMe].killsThisLife, true, Scores.ScoreData[player].lastPersonWhoDamagedMe);
                                 //canAssignMutator = true;
                                 break;
                             case AliveState.KillSteal: // Damaged recently, but took damage from someone else recently too.
@@ -2910,7 +2908,6 @@ public void DoUpdate()
                                 {
                                     statusMessage += Scores.ScoreData[player].lastPersonWhoDamagedMe + " (KILL STEAL!)";
                                 }
-                                PlayAnnouncer(Scores.ScoreData[Scores.ScoreData[player].lastPersonWhoDamagedMe].killsThisLife, false, Scores.ScoreData[player].lastPersonWhoDamagedMe);
                                 //canAssignMutator = true;
                                 break;
                             case AliveState.AssistedKill: // Assist (not damaged recently or GM kill).
@@ -3080,9 +3077,10 @@ void ApplyOnKillMutator(string player)
                 if (loadedVessels.Current == null || !loadedVessels.Current.loaded || VesselModuleRegistry.ignoredVesselTypes.Contains(loadedVessels.Current.vesselType))
                     continue;
                 var craftName = loadedVessels.Current.GetName();
+                if (!Scores.Players.Contains(craftName)) continue;
                 if (BDArmorySettings.MUTATOR_APPLY_GUNGAME && Scores.ScoreData[player].aliveState == AliveState.AssistedKill && Scores.ScoreData[player].everyoneWhoDamagedMe.Contains(craftName))
                     SpawnUtils.ApplyMutators(loadedVessels.Current, true); // Reward everyone involved on assists.
-                else if (Scores.ScoreData[player].lastPersonWhoDamagedMe == craftName || (Scores.ScoreData[player].aliveState == AliveState.KillSteal && Scores.ScoreData[player].previousPersonWhoDamagedMe == craftName))
+                else if (Scores.ScoreData[player].lastPersonWhoDamagedMe == craftName || (BDArmorySettings.MUTATOR_APPLY_GUNGAME && Scores.ScoreData[player].aliveState == AliveState.KillSteal && Scores.ScoreData[player].previousPersonWhoDamagedMe == craftName))
                     SpawnUtils.ApplyMutators(loadedVessels.Current, true); // Reward clean kills and those whom have had their kills stolen.
                 else continue;
                 if (BDArmorySettings.DEBUG_COMPETITION) Debug.Log($"[BDArmory.BDACompetitionMode:{CompetitionID}]: Assigning On Kill mutator for {player} to {craftName}");
diff --git a/BDArmory/Competition/BDATournament.cs b/BDArmory/Competition/BDATournament.cs
index 51203e876..a5b143302 100644
--- a/BDArmory/Competition/BDATournament.cs
+++ b/BDArmory/Competition/BDATournament.cs
@@ -1450,6 +1450,7 @@ IEnumerator RunTournamentCoroutine()
                     message = $"Running heat {heatIndex} of round {roundIndex} of tournament {tournamentState.tournamentID} ({heatsRemaining} heats remaining in the tournament).";
                     BDACompetitionMode.Instance.competitionStatus.Add(message);
                     Debug.Log("[BDArmory.BDATournament]: " + message);
+                    
                     if (firstRun) SpawnUtilsInstance.Instance.gunGameProgress.Clear(); // Clear gun-game progress.
                     int attempts = 0;
                     bool unrecoverable = false;
diff --git a/BDArmory/Competition/Scoring.cs b/BDArmory/Competition/Scoring.cs
index 235784d09..eb6c4d49c 100644
--- a/BDArmory/Competition/Scoring.cs
+++ b/BDArmory/Competition/Scoring.cs
@@ -521,7 +521,12 @@ public bool RegisterDeath(string vesselName, GMKillReason gmKillReason = GMKillR
                 else // Last hit from someone else was recent => Kill Steal
                 { ScoreData[vesselName].aliveState = AliveState.KillSteal; }
 
-                if (Players.Contains(ScoreData[vesselName].lastPersonWhoDamagedMe)) ++ScoreData[ScoreData[vesselName].lastPersonWhoDamagedMe].killsThisLife;
+                if (Players.Contains(ScoreData[vesselName].lastPersonWhoDamagedMe))
+                {
+                    ++ScoreData[ScoreData[vesselName].lastPersonWhoDamagedMe].killsThisLife;
+                    BDACompetitionMode.Instance.PlayAnnouncer(ScoreData[ScoreData[vesselName].lastPersonWhoDamagedMe].killsThisLife, false, ScoreData[vesselName].lastPersonWhoDamagedMe);
+                }
+
                 if (BDArmorySettings.REMOTE_LOGGING_ENABLED)
                 { BDAScoreService.Instance.TrackKill(ScoreData[vesselName].lastPersonWhoDamagedMe, vesselName); }
             }
diff --git a/BDArmory/Distribution/GameData/BDArmory/ChangeLog.txt b/BDArmory/Distribution/GameData/BDArmory/ChangeLog.txt
index 8910eeeec..89c31df8b 100644
--- a/BDArmory/Distribution/GameData/BDArmory/ChangeLog.txt
+++ b/BDArmory/Distribution/GameData/BDArmory/ChangeLog.txt
@@ -1,8 +1,8 @@
-
+﻿
 ﻿IMPROVEMENTS / FIXES
 - General:
        - Fix false positives of Krakensbane being active on vessel switches.
-       - Add GUn Game game option to the Mutator GameMode - craft progress through a list of mutators (weapon changes or otherwise) every time they score a kill. Ideally used with Continuous Spawn.
+       - Add Gun Game game option to the Mutator GameMode - craft progress through a list of mutators (weapon changes or otherwise) every time they score a kill. Ideally used with Continuous Spawn.
 - AI / WM:
        - Clamp the vertical adjustment of the fly-to position to at most half the distance to the target (should fix the issue of being unable to attack targets directly below).
        - Add a slider for biasing the Immelmann pitch direction when not close to terrain/min altitude.
diff --git a/BDArmory/Settings/RWPSettings.cs b/BDArmory/Settings/RWPSettings.cs
index cea77218d..5b78bccab 100644
--- a/BDArmory/Settings/RWPSettings.cs
+++ b/BDArmory/Settings/RWPSettings.cs
@@ -55,7 +55,7 @@ public class RWPSettings
         {"MUTATOR_MODE", true},
         {"MUTATOR_DURATION", 0},
         {"MUTATOR_APPLY_KILL", true},
-        {"MUTATOR_APPLY_GUNGAME", false},
+        {"MUTATOR_APPLY_GUNGAME", true},
         {"MUTATOR_APPLY_GLOBAL", false},
         {"MUTATOR_APPLY_TIMER", false},
         {"MUTATOR_APPLY_NUM", 1},
diff --git a/BDArmory/UI/BDArmorySetup.cs b/BDArmory/UI/BDArmorySetup.cs
index 2de5dad12..15927a96c 100644
--- a/BDArmory/UI/BDArmorySetup.cs
+++ b/BDArmory/UI/BDArmorySetup.cs
@@ -3379,7 +3379,7 @@ void WindowSettings(int windowID)
                         BDArmorySettings.MUTATOR_APPLY_KILL = GUI.Toggle(SRightRect(line, 1f), BDArmorySettings.MUTATOR_APPLY_KILL, StringUtils.Localize("#LOC_BDArmory_Settings_MutatorKill"));
                         if (BDArmorySettings.MUTATOR_APPLY_KILL) // if more than 1 mutator selected, will randomly assign mutator on kill
                         {
-                            BDArmorySettings.MUTATOR_APPLY_KILL = GUI.Toggle(SLeftRect(++line, 1f), BDArmorySettings.MUTATOR_APPLY_GUNGAME, StringUtils.Localize("#LOC_BDArmory_Settings_MutatorGungame"));
+                            BDArmorySettings.MUTATOR_APPLY_GUNGAME = GUI.Toggle(SLeftRect(++line, 1f), BDArmorySettings.MUTATOR_APPLY_GUNGAME, StringUtils.Localize("#LOC_BDArmory_Settings_MutatorGungame"));
                             BDArmorySettings.MUTATOR_APPLY_GLOBAL = false;
                             BDArmorySettings.MUTATOR_APPLY_TIMER = false;
                             BDArmorySettings.GG_PERSISTANT_PROGRESSION = GUI.Toggle(SLeftRect(++line), BDArmorySettings.GG_PERSISTANT_PROGRESSION, StringUtils.Localize("#LOC_BDArmory_settings_gungame_progression"));
diff --git a/BDArmory/Utils/SoundUtils.cs b/BDArmory/Utils/SoundUtils.cs
index a01c3f227..a63185db9 100644
--- a/BDArmory/Utils/SoundUtils.cs
+++ b/BDArmory/Utils/SoundUtils.cs
@@ -17,6 +17,7 @@ public static class SoundUtils
         /// If it's not in the cache, then load the audioclip from the GameDatabase and cache it for future use.
         /// </summary>
         /// <param name="soundPath">The path to a valid audioclip.</param>
+        /// <param name="allowMissing">Don't log an error if the sound file doesn't exist, just log it instead.</param>
         /// <returns>The AudioClip.</returns>
         public static AudioClip GetAudioClip(string soundPath, bool allowMissing = false)
         {
```

I've tested it and S6R1 seems to be behaving correctly with these changes now.

**Pull request merged**